### ISSUE TITLE
Plural forms generator

### ISSUE THREAD

**Issue by ganchenkor** [i18next/i18next#280]
As I understand many languages have many plural forms. How can I generate proper number of plural keys for each language?
Example:
en/translation.json
"car": "**count** car",
"car_plural": "**count** cars"

ru/translation.json
"car": "**count** машина"
"car_plural_2": "**count машины"
"car_plural_5": "__count** машин"

I know this because I'm Russian, but how can I make sure on which plural number for each language using our http://docs.translatehouse.org/projects/localization-guide/en/latest/l10n/pluralforms.html?id=l10n/pluralforms rules

Is there any generator like it reads the keys, and generates keys with proper number of plural forms for each key producing accurate json which later could be sent to some service like gengo.com(anybody has experience with them?) (For example: arabic has 6 plural forms, and I have no idea by which number and I want translator to keep in mind for that, so he won't just translate 1 form of plural, apparently I want all of them)
Does transifex.com can help me with that? Or this is just like github for translations?

Please let me know if this problem is already solved

Example of the tool would be:
1) we create source translator file
2) we provide which languages we want to generate
3) it detects if we have any key with plural at the end
4)it uses pluralforms rules to produce correct number of keys for each language

**Comment by the-teacher** [comment]
@ganchenkor 
@jamuhl 

For lang related pluralization you can use official rules from here

http://unicode.org/repos/cldr-tmp/trunk/diff/supplemental/language_plural_rules.html

for RU lang it's not a problem to build following helper method

```
var ru_plural = function(k, n){
  var postfix = '_other';

  var a = (n % 10 === 1);
  var b = (n % 100 !== 11);
  if( a && b ){ postfix = '_one' }

  var c = [2, 3, 4].indexOf(n % 10)     !== -1;
  var d = [12, 13, 14].indexOf(n % 100) === -1;
  if( c && d ){ postfix = '_few' }

  var e = n % 10 === 0;
  var f = [5, 6, 7, 8, 9].indexOf(n % 10)   !== -1;
  var g = [11, 12, 13, 14].indexOf(n % 100) !== -1;
  if( e || f || g ){ postfix = '_many' }

  return k + postfix;
}
```

Now we can use

```
i18next.t(ru_plural('currencyRub', 0))
"рублей"
i18next.t(ru_plural('currencyRub', 1))
"рубль"
i18next.t(ru_plural('currencyRub', 2))
"рубля"
i18next.t(ru_plural('currencyRub', 3))
"рубля"
i18next.t(ru_plural('currencyRub', 4))
"рубля"
i18next.t(ru_plural('currencyRub', 5))
"рублей"
i18next.t(ru_plural('currencyRub', 0.5))
"рубля"
```

**Comment by jamuhl** [comment]
@the-teacher its not about the way do plurals...thats already handled: http://i18next.com/translate/pluralMulti/

it's more about using translations tools (like eg. passalo,...) where the translation tool does not know how to handle pluralforms inside json: `en: 'bla_plural'` > `ru: 'bla_0, bla_1,...'`. basically they just generate a `ru: bla_plural` so support for plurals is more or less impossible on such tools.

at least for json most tools don't respect the syntactic sugar to handle plurals...but most might get i18next.json(en) -> gettext(en) -> tool -> gettext(ru) -> i18next.json(ru) right.

**Comment by the-teacher** [comment]
@jamuhl thanks, for your note. I saw, and I saw a source code. I just show another way to do that.

Btw, I surprised with  syntax of pluralization rules 'bla_0, bla_1` , be course I used to use rails way of pluralization files when it's provides with prefixes/postfixes `zero`, `one`, `few`, `many`, `other`. I think it's a common way be cause it described on http://unicode.org .  Probably, I wrong. It's my personal prefer.
